base/src/com/google/idea/blaze/base/bazel/LocalInvoker.java

```diff
@@ -0,0 +1,275 @@
+    super(project, blazeContext, buildSystem, binaryPath(binaryType, project));
+    this.buildBinaryType = binaryType;
+  }
+
+  @Override
+  public BuildResultHelper createBuildResultHelper() {
+    return new BuildResultHelperBep();
+  }
+
+  @Override
+  public BlazeCommandRunner getCommandRunner() {
+    return new CommandLineBlazeCommandRunner();
+  }
+
+  @Override
+  public boolean supportsHomeBlazerc() {
+    return true;
+  }
+
+  @Override
+  public boolean supportsParallelism() {
+    return false;
+  }
+
+  @Override
+  public ImmutableSet<Capability> getCapabilities() {
+    return ImmutableSet.of(Capability.IS_LOCAL, Capability.SUPPORTS_CLI);
+  }
+
+  @Override
+  public BuildEventStreamProvider invoke(BlazeCommand.Builder blazeCommandBuilder)
+    throws BuildException {
+    try {
+      performGuardCheck(project, blazeContext);
+    }
+    catch (ExecutionDeniedException e) {
+      throw new BuildException(e.getMessage(), e);
+    }
+    File outputFile = BuildEventProtocolUtils.createTempOutputFile();
+    BuildResult buildResult =
+      issueBuild(blazeCommandBuilder, WorkspaceRoot.fromProject(project), blazeContext, outputFile);
+    if (blazeCommandBuilder.build().getName().equals(BlazeCommandName.BUILD)) {
+      BuildDepsStatsScope.fromContext(blazeContext)
+        .ifPresent(stats -> stats.setBazelExitCode(buildResult.exitCode));
+    }
+    return getBepStream(outputFile);
+  }
+
+  @Override
+  public InputStream invokeQuery(BlazeCommand.Builder blazeCommandBuilder) {
+    try {
+      performGuardCheck(project, blazeContext);
+    }
+    catch (ExecutionDeniedException e) {
+      logger.error(e);
+      return null;
+    }
+
+    BlazeCommand blazeCommand = blazeCommandBuilder.build();
+    try (Closer closer = Closer.create()) {
+      Path tempFile =
+        Files.createTempFile(
+          String.format("intellij-bazel-%s-", blazeCommand.getName()), ".stdout");
+      OutputStream out = closer.register(Files.newOutputStream(tempFile));
+      WorkspaceRoot workspaceRoot = WorkspaceRoot.fromProject(project);
+      Function<String, String> rootReplacement =
+        WorkspaceRootReplacement.create(workspaceRoot.path(), blazeCommand);
+      boolean isUnitTestMode = ApplicationManager.getApplication().isUnitTestMode();
+      int retVal =
+        ExternalTask.builder(workspaceRoot)
+          .addBlazeCommand(blazeCommand)
+          .context(blazeContext)
+          .stdout(out)
+          .stderr(
+            LineProcessingOutputStream.of(
+              line -> {
+                line = rootReplacement.apply(line);
+                // errors are expected, so limit logging to info level
+                if (isUnitTestMode) {
+                  // This is essential output in bazel-in-bazel tests if they fail.
+                  System.out.println(line.stripTrailing());
+                }
+                Logger.getInstance(this.getClass()).info(line.stripTrailing());
```

Comment:
The value of `line.stripTrailing()` can be stored in a variable to avoid repeated calls.

Reply:
Let's not deviating from AOSP for this either.